snapshot.go

```diff
@@ -115,6 +115,10 @@ func getTxFromSnapshotFileName(fileName string) (uint64, error) {
 // started (i.e. no other snapshot was in progress). When the snapshot
 // goroutine successfully completes a snapshot, onSuccess is called.
 func (db *DB) asyncSnapshot(ctx context.Context, onSuccess func()) {
+	db.snapshot(ctx, true, onSuccess)
+}
+
+func (db *DB) snapshot(ctx context.Context, async bool, onSuccess func()) {
```

Comment:
Could we return an error from snapshot and move the callback to just the async version? Not sure if it's too complex with the txn beginning

Reply:
The refactor gets a little hairy with the tx and metadata. It's gonna require a bigger refactor than I think is worth it right now.